productions/p1.py

```diff
@@ -86,4 +86,5 @@ def apply(self, graph: HyperGraph) -> HyperGraph:
             graph.add_hyperedge([node_3, node_4, reversed_node_map[left_side.nodes[3]], node_center])
             graph.add_hyperedge([node_4, node_1, reversed_node_map[left_side.nodes[0]], node_center])
 
-        return graph
+            return graph
+        return None
```

Comment:
I am not fan of that, in the future we will need to pipe graph through production after production. It is reasonable to return graph that can go straight into next production